tests/unit/api/bucketPutCors.js

```diff
@@ -0,0 +1,261 @@
+                rule[elementArr].forEach(value => {
+                    xml.push(`<${element}>${value}</${element}>`);
+                });
+            }
+        });
+        if (rule.id) {
+            xml.push(`<ID>${rule.id}</ID>`);
+        }
+        if (rule.maxAgeSeconds && !Array.isArray(rule.maxAgeSeconds)) {
+            xml.push(`<MaxAgeSeconds>${rule.maxAgeSeconds}</MaxAgeSeconds>`);
+        }
+        xml.push('</CORSRule>');
+    });
+    xml.push('</CORSConfiguration>');
+    return xml.join('');
+}
+
+function _testPutBucketCors(authInfo, request, log, errCode, cb) {
+    bucketPutCors(authInfo, request, log, err => {
+        assert(err, 'Expected err but found none');
+        assert.deepStrictEqual(err, errors[errCode]);
+        cb();
+    });
+}
+
+function _generateSampleXml(value) {
+    const xml = '<CORSConfiguration>' +
+    '<CORSRule>' +
+    '<AllowedMethod>PUT</AllowedMethod>' +
+    '<AllowedOrigin>www.example.com</AllowedOrigin>' +
+    `${value}` +
+    '</CORSRule>' +
+    '</CORSConfiguration>';
+
+    return xml;
+}
+
+describe('putBucketCORS API', () => {
+    beforeEach(done => {
+        cleanup();
+        bucketPut(authInfo, testBucketPutRequest,
+        locationConstraint, log, done);
+    });
+    afterEach(() => cleanup());
+
+    it('should update a bucket\'s metadata with cors resource', done => {
+        const sampleXml = _constructCorsXml(sampleCors);
+        const testBucketPutCorsRequest = _createPutCorsRequest(sampleXml);
+        bucketPutCors(authInfo, testBucketPutCorsRequest, log, err => {
+            if (err) {
+                process.stdout.write(`Err putting website config ${err}`);
+                return done(err);
+            }
+            return metadata.getBucket(bucketName, log, (err, bucket) => {
+                if (err) {
+                    process.stdout.write(`Err retrieving bucket MD ${err}`);
+                    return done(err);
+                }
+                const uploadedCors = bucket.getCors();
+                assert.deepStrictEqual(uploadedCors, sampleCors);
+                return done();
+            });
+        });
+    });
+
+    it('should return BadDigest if md5 is omitted', done => {
+        const sampleXml = _constructCorsXml(sampleCors);
+        const testBucketPutCorsRequest = _createPutCorsRequest(sampleXml);
+        testBucketPutCorsRequest.headers['content-md5'] = undefined;
+        _testPutBucketCors(authInfo, testBucketPutCorsRequest,
+            log, 'BadDigest', done);
+    });
+
+    it('should return MalformedXML if body greater than 64KB', done => {
+        const body = Buffer.alloc(65537); // 64 * 1024 = 65536 bytes
+        const testBucketPutCorsRequest = _createPutCorsRequest(body);
+        _testPutBucketCors(authInfo, testBucketPutCorsRequest,
+            log, 'MalformedXML', done);
+    });
+
+    it('should return InvalidRequest if more than one MaxAgeSeconds', done => {
+        const testCors = sampleCors.slice();
```

Comment:
Ah, nice! Why not pass the copy in as you had done before, and keep `_constructCorsXml` the same? Maybe you are planning to reuse it for other customized objects? It's very minor, but that would obviate the step for those other tests that don't need to create a copy.